File: csharp/CSharp/CSharpParserBase.cs
using System;
using System.Collections.Generic;
using System.IO;
using Antlr4.Runtime;

public abstract class CSharpParserBase : Parser
{
    protected CSharpParserBase(ITokenStream input)
        : base(input)
    {
    }

    protected CSharpParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput)
    {
    }

    protected bool IsLocalVariableDeclaration()
    {
        var local_var_decl = this.Context as CSharpParser.Local_variable_declarationContext;
        if (local_var_decl == null) return true;
        var local_variable_type = local_var_decl.local_variable_type();
        if (local_variable_type == null) return true;
        if (local_variable_type.GetText() == "var") return false;
        return true;
    }
}

File: csharp/Java/CSharpParserBase.java
import org.antlr.v4.runtime.*;
import java.util.Stack;

public abstract class CSharpParserBase extends Parser
{
    protected CSharpParserBase(TokenStream input)
    {
        super(input);
    }

    protected boolean IsLocalVariableDeclaration()
    {
        if (!(this._ctx instanceof CSharpParser.Local_variable_declarationContext)) {
            return false;
        }
        CSharpParser.Local_variable_declarationContext local_var_decl = (CSharpParser.Local_variable_declarationContext)this._ctx;
        if (local_var_decl == null) return true;
        CSharpParser.Local_variable_typeContext local_variable_type = local_var_decl.local_variable_type();
        if (local_variable_type == null) return true;
        if (local_variable_type.getText().equals("var")) return false;
        return true;
    }
}
